### ISSUE TITLE
Problem with latest docker image: "failed to get urandom"

### ISSUE THREAD

**Issue by gingerbeardman** [huginn/huginn#3262]
back on 13th May my docker install of huginn was upgraded by Watchtower to latest.

It will no longer launch properly.

```
rake aborted!
foreman stderr | failed to get urandom
```

- I'm running it on a Synology DSM 6.2.3 if that makes any difference.

**Comment by gingerbeardman** [comment]
I used Portainer to

1. pull `huginn/huginn:45378efa7f3a77a8589378d5948a47544c60e2a7`
2. recreate the container with the same settings using that older version

it's now working again.

I'll prevent watchtower from updating it until i hear the all clear in this issue.

**Comment by gingerbeardman** [comment]
@knu FYI

**Comment by gingerbeardman** [comment]
Any thoughts @knu ?

**Comment by knu** [comment]
The message suggests the application couldn't open or read from /dev/urandom, I guess?

The app has a low limit of maximum open files (typically raised by `ulimit`) or the system is short of random entropy.

**Comment by gingerbeardman** [comment]
Yes, I think it's the later. The kernel used on my Synology does not support it.

Of course, Huginn docker has been working fine until 13th May.

The breaking change was introduced in the container released on 12th May.

https://hub.docker.com/layers/huginn/huginn/2a04d2b071ab003bd24fc9a6ce3af9108ca101dd/images/sha256-70ea35ef79c37740bd721cf1da785054e028055caaf71fd4ee2d514da55e71ed?context=explore

I've encountered exactly this breaking change with another docker container, I'll try to remember which one, and will reply with any further details.

Let me know if you have any idea of what changed in 12th May release that would cause this.

**Comment by gingerbeardman** [comment]
It seems it may be fixable with either dependency or docker config change.

Related 
- https://github.com/stefansundin/rssbox/issues/67 (missed dependency)
- https://github.com/nextcloud/docker/issues/1574#issuecomment-1048244349 (my comment)

**Comment by gingerbeardman** [comment]
I'm looking at the biggest differences in the 12th May container (vs May 8th container)
- `LABEL org.opencontainers.image.version=22.04`
  - previously 18.04
- `BASE_IMAGE_TAG=jammy RUBY_VERSION=3.2.2`
  - previously 2.7

**Comment by dsander** [comment]
I don't really see what we can do about this, yes we moved to a current ubuntu docker base image and ruby version recently, but it sounds to me the  bug isn't directly caused by that but by either the really old kernel or linux configuration synology DSM uses.

It sounds like https://github.com/ruby/ruby/pull/7705 works around it, so if [this commit](https://github.com/ruby/ruby/commit/64e503eb62aff0952b655e9a86217e355f786146) makes it into the next ruby release (which hopefully gets backported into 3.2), we can bump  the ruby version.

**Comment by gingerbeardman** [comment]
Thanks. 

I'm sure there are many Linux servers with old Kernel still out there, not just DSM.

I'll keep an eye on the Ruby change!

**Comment by gingerbeardman** [comment]
> but it sounds to me the bug isn't directly caused by that but by either the really old kernel or linux configuration synology DSM uses.

I've been thinking more about this and looking into it more. One could argue that it is a bug, in that the developer of whichever tool was not aware that /dev/random is not guaranteed to be available. As seen in the Ruby/OpenSSL change.

**Comment by hihouhou** [comment]
@gingerbeardman about docker image with ruby 3.2 https://github.com/hihouhou/docker-huginn/commit/f47d2cdf81873f6c6afc4c80e7c1d300ea6be37b (if it can help you)

**Comment by gingerbeardman** [comment]
@hihouhou thanks I'll try it soon

**Comment by gingerbeardman** [comment]
I never got round to trying it.

**Comment by dissidente** [comment]
Can I get an 'up' here?
Frozen at an old version because of this issue =|

**Comment by gingerbeardman** [comment]
@dissidente did you try the recommended version above? you can install to new docker container to check if it works, no need to disrupt your setup.

personally, i have no need to move to a more recent version so i never checked

**Comment by gingerbeardman** [comment]
> @gingerbeardman about docker image with ruby 3.2 [hihouhou/docker-huginn@f47d2cd](https://github.com/hihouhou/docker-huginn/commit/f47d2cdf81873f6c6afc4c80e7c1d300ea6be37b) (if it can help you)

What is the easiest way for me to try this? I'm actually quite inexperienced with the finer points of docker.

**Comment by hihouhou** [comment]
you need to clone https://github.com/hihouhou/docker-huginn locally , update the .env and build it -> docker build  -t huginn:latest . and start the container

**Comment by jjeffers** [comment]
@gingerbeardman Yeah, I'll remove the tag.
